neo4j/_async/driver.py

```diff
@@ -392,6 +392,108 @@ async def supports_multi_db(self):
             await session._connect(READ_ACCESS)
             return session._connection.supports_multiple_databases
 
+    async def query(self, query, parameters=None, **kwargs):
+        """
+        Run a Cypher query within an managed transaction and
+        all the retries policy will be applied.
+
+        The query is sent and the result header received
+        immediately and the :class:`neo4j.QueryResult`is
+        fetched.
+
+        For more usage details, see :meth:`.AsyncTransaction.query`.
+
+        For auto-commit queries, use :meth:`AsyncSession.run`.
+
+        For access to the :class:`neo4j.AsyncResult` object,
+        use :meth:`AsyncDriver.execute` and :meth:`.AsyncTransaction.run`
+
+        :param query: cypher query
+        :type query: str, neo4j.Query
+        :param parameters: dictionary of parameters
+        :type parameters: dict
+        :param kwargs: additional keyword parameters
+
+        :returns: a new :class:`neo4j.QueryResult` object
+        :rtype: QueryResult
+        """
+        session_kwargs = {}
+        if "database" in kwargs:
```

Comment:
The doc string doesn't mention anywhere, that the `database` kwarg has a special function and will not get used as a query parameter. Maybe, `database=None` should be an explicit keyword argument, as in

```python
async def query(self, query, parameters=None, database=None, **kwargs):
```

Tbf, this also applies to `driver.session`. It does not specify what config options there are, but at least there is an explicit list of available config options in the API docs https://github.com/neo4j/neo4j-python-driver/blob/5.0/docs/source/api.rst#session-configuration